service/storage_service.cc

```diff
@@ -772,6 +774,9 @@ future<> storage_service::sstable_cleanup_fiber(raft::server& server, sharded<se
 
                 tasks.reserve(keyspaces.size());
 
+                co_await _db.invoke_on_all([&] (replica::database& db) {
+                    return db.flush_all_tables();
+                });
```

Comment:
@gleb-cloudius: in e277ec6aef97eba8b78616ad2fe1245961544f22 we skip and keyspace that has a local replication strategy, even if it's not a system keyspace.
I think it's a more accurate requirement for cleanup.

Reply:
> @gleb-cloudius: in [e277ec6](https://github.com/scylladb/scylladb/commit/e277ec6aef97eba8b78616ad2fe1245961544f22) we skip and keyspace that has a local replication strategy, even if it's not a system keyspace. I think it's a more accurate requirement for cleanup.

Yes. But do we have those?

Reply:
I think it's unlikely to have such user tables (maybe a user keyspace can be created with LocalStrategy, not sure), but it would be future compatible in case we have other keyspaces that use local replication that are not considered by `is_system_keyspace`, e.g. if we replace system_distributed with consistent raft-managed tables that are local to each node.

Reply:
The point it that the criterion for cleanup should consider use of vnodes vs. tablets and the keyspace replication strategy rather than inferring the replication strategy from being a system keyspace or not.

Reply:
The code here needs to use `get_non_local_strategy_keyspaces_erms` instead of do its own filtering, but what it has to do with this series?

Reply:
It's tangential indeed.

Reply:
@gleb-cloudius i can help address it in a follow-up change. what do you think?

Reply:
It would be great!

Reply:
i created #16974 to track this task.